src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java

```diff
@@ -106,4 +114,16 @@ public JLabel getEmptyTextLabel() {
     return emptyText;
   }
 
+  private class NoTokenMouseAdapter extends MouseAdapter {
+    @Override
+    public void mouseClicked(MouseEvent e) {
+      if (exerciseGroupsTree.isEmpty()
+          && exerciseGroupsTree.getEmptyText().getText().contains(
+              getText("ui.exercise.ExercisesView.setToken"))) {
+        DataContext context = DataManager.getInstance().getDataContext(e.getComponent());
+        ActionUtil.launch(APlusAuthenticationAction.ACTION_ID, context);
+      }
```

Comment:
Is there a need for consuming the event?

Reply:
I'm not sure about this, but I don't think so